### PR TITLE
Hopefully fix `TestMount` failures

### PR DESCRIPTION
**Pull request by MichaelEischer**
What does this PR change? What problem does it solve?
-----------------------------------------------------
The `TestMount` test fails from time to time. One instance of this is shown in https://github.com/restic/restic/actions/runs/4968574603/jobs/8891251180:
```
2023/05/13 19:42:24 restic/cmd_mount.go:178	restic.runMount.func2	163	fuse: <- Lookup [ID=0x4 Node=0x1 Uid=1001 Gid=123 Pid=10047] "snapshots"
2023/05/13 19:42:24 fuse/snapshots_dir.go:96	fuse.(*SnapshotsDir).Lookup	163	Lookup(snapshots)
2023/05/13 19:42:24 restic/cmd_mount.go:178	restic.runMount.func2	164	fuse: <- Interrupt [ID=0x5 Node=0x0 Uid=0 Gid=0 Pid=0] ID 0x4
2023/05/13 19:42:24 logger/log.go:59	logger.(*Backend).List	159	List(snapshot)
2023/05/13 19:42:24 restic/cmd_mount.go:178	restic.runMount.func2	164	fuse: -> [ID=0x5] Interrupt
2023/05/13 19:42:24 logger/log.go:61	logger.(*Backend).List	159	  list err <nil>
2023/05/13 19:42:24 repository/repository.go:585	repository.(*Repository).LoadIndex	163	Loading index
2023/05/13 19:42:24 restic/cmd_mount.go:178	restic.runMount.func2	163	fuse: -> [ID=0x4] Lookup error=EIO: Fatal: context canceled
2023/05/13 19:42:24 restic/cmd_mount.go:178	restic.runMount.func2	165	fuse: <- Lookup [ID=0x6 Node=0x1 Uid=1001 Gid=123 Pid=10047] "snapshots"
2023/05/13 19:42:24 fuse/snapshots_dir.go:96	fuse.(*SnapshotsDir).Lookup	165	Lookup(snapshots)
```

There `repository.LoadIndex` in canceled while it is in progress, which results in a `Fatal: context canceled` error. The problem there is that the `Fatal` error type only contains a string which makes it impossible to properly check for the contained `Context canceled` error.

Thus, this PR removes most usages of the `Fatal` error type from the `internal` packages. This also fixes a few cases of Fatal errors within another Fatal error.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Follow up to #4317.

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- ~~[ ] I have added documentation for relevant changes (in the manual).~~
- ~~[ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).~~
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: internal/backend/rest/rest.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func Create(ctx context.Context, cfg Config, rt http.RoundTripper) (*Backend, er
 
 	_, err = be.Stat(ctx, restic.Handle{Type: restic.ConfigFile})
 	if err == nil {
-		return nil, errors.Fatal("config file already exists")
+		return nil, fmt.Errorf("config file already exists")
```
Comment (greatroar) [inline internal/backend/rest/rest.go:65]:
```suggestion
		return nil, errors.New("config file already exists")
```

File: internal/repository/repack.go [hunk 2]
```diff
@@ -29,7 +29,7 @@ func Repack(ctx context.Context, repo restic.Repository, dstRepo restic.Reposito
 	debug.Log("repacking %d packs while keeping %d blobs", len(packs), keepBlobs.Len())
 
 	if repo == dstRepo && dstRepo.Connections() < 2 {
-		return nil, errors.Fatal("repack step requires a backend connection limit of at least two")
+		return nil, fmt.Errorf("repack step requires a backend connection limit of at least two")
```
Comment (greatroar) [inline internal/repository/repack.go:32]:
```suggestion
		return nil, errors.New("repack step requires a backend connection limit of at least two")
```

File: internal/repository/repository.go [hunk 3]
```diff
@@ -613,7 +613,7 @@ func (r *Repository) LoadIndex(ctx context.Context) error {
 			}
 		})
 		if invalidIndex {
-			return errors.Fatal("index uses feature not supported by repository version 1")
+			return fmt.Errorf("index uses feature not supported by repository version 1")
```
Comment (greatroar) [inline internal/repository/repository.go:616]:
```suggestion
			return errors.New("index uses feature not supported by repository version 1")
```

File: internal/restic/snapshot_group.go [hunk 4]
```diff
@@ -26,7 +25,7 @@ func splitSnapshotGroupBy(s string) (SnapshotGroupByOptions, error) {
 			l.Tag = true
 		case "":
 		default:
-			return SnapshotGroupByOptions{}, errors.Fatal("unknown grouping option: '" + option + "'")
+			return SnapshotGroupByOptions{}, fmt.Errorf("unknown grouping option: '%v'", option)
```
Comment (greatroar) [inline internal/restic/snapshot_group.go:28]:
```suggestion
			return SnapshotGroupByOptions{}, fmt.Errorf("unknown grouping option: %q", option)
```

File: internal/repository/repository.go [hunk 5]
```diff
@@ -110,16 +110,16 @@ func (c *CompressionMode) Type() string {
 // New returns a new repository with backend be.
 func New(be restic.Backend, opts Options) (*Repository, error) {
 	if opts.Compression == CompressionInvalid {
-		return nil, errors.Fatalf("invalid compression mode")
+		return nil, fmt.Errorf("invalid compression mode")
```
[the remark this answers is not in the document]
Comment (MichaelEischer) [reply internal/repository/repository.go]: Yes, it's more like an additional sanity check to ensure that this won't break in the future.

File: internal/repository/repack.go [hunk 6]
```diff
@@ -2,10 +2,10 @@ package repository
 
 import (
 	"context"
+	"fmt"
```
Comment (greatroar) [inline internal/repository/repack.go]: Oops:

```suggestion
	"errors"
```

**Comment by MichaelEischer** [comment]
I've fixed the nitpicks.

**Pull request merged**